### PR TITLE
504 fix ci and modernize

### PR DESCRIPTION
**Pull request by fenekku**
- closes https://github.com/galterlibrary/Prism/issues/504
- modernize package + use galter-subjects-utils
- release: v2023.11.06

### CODE REVIEW

**Review by carsonicator — APPROVED** [review]
Thanks @fenekku

**Review by Meowcenary — APPROVED** [review]
Looks good to me! I don't think I totally get it, but the broad strokes made sense.

File: README.md [hunk 1]
```diff
@@ -1,58 +1,102 @@
...
+For instance administrators, after you have installed the extension as per the steps above, you will want to reload your instance's fixtures: `pipenv run invenio rdm-records fixtures`. This will install the new terms in your instance.
 
-### TODO -- Maintainers
+Updating existing terms currently requires manual replacement.
 
-When a new list of LCSH term comes out, this package should be updated. Here we show how.
+### Maintainers
 
-0- Make sure you have cloned this package. Then install it locally with:
+When a new list of MeSH term comes out, this package should be updated. Here we show how.
```
Comment (sharpattack) [inline README.md:39]: Shouldn't this say `new list of LCSH terms`?

File: README.md [hunk 2]
```diff
@@ -1,58 +1,102 @@
...
 
 There are 2 types of users for this package. Maintainers of the package and instance administrators.
 
 ### Instance administrators
 
-For instance administrators, after you have installed the extension as per the steps above, you are done. It should just work.
+For instance administrators, after you have installed the extension as per the steps above, you will want to reload your instance's fixtures: `pipenv run invenio rdm-records fixtures`. This will install the new terms in your instance.
 
-### TODO -- Maintainers
+Updating existing terms currently requires manual replacement.
```
Comment (sharpattack) [inline README.md:35]: Is manual replacement straightforward? I am not sure what that looks like, so am just wondering if instructions could be added?

File: README.md [hunk 3]
```diff
@@ -1,58 +1,102 @@
...
+```bash
+pipenv run pip install -e .[dev]
+```
 
-    make update
+1. Use the installed `galter-subjects-utils` tool to get the new list:
+
+```bash
+pipenv run galter-subjects-utils lcsh --output-file invenio_subjects_lcsh/vocabularies/subjects_lcsh.jsonl
+```
```
Comment (sharpattack) [inline README.md:51]: `subjecrts_lcsh.jsonl` will be a relatively large file, no? Just something to keep an eye on if it is being committed in the project.

**Review by sharpattack — APPROVED** [review]
Just some minor comments, but looks good to me.

File: .github/workflows/pypi-publish.yml [hunk 4]
```diff
@@ -5,27 +5,34 @@ on:
...
+          invoke check-manifest
 
       - name: Build package
         run: |
-          python setup.py sdist bdist_wheel
+          python -m build .
 
       - name: Publish to PyPI
-        uses: pypa/gh-action-pypi-publish@v1.6.4
+        uses: pypa/gh-action-pypi-publish@release/v1
```
[the remark this answers is not in the document]
Comment (fenekku) [reply .github/workflows/pypi-publish.yml:35]: Yes which is why we have: https://github.com/galterlibrary/invenio-subjects-lcsh/blob/master/.github/dependabot.yml#L18 to automate the updating of GitHub actions :stuck_out_tongue_winking_eye:

File: README.md (hunk 3 above)
Comment (fenekku) [reply README.md:51]: Oh, I mean it is a large file (~60MB), it is already included in the project and it was updated in this PR :smile: . There isn't a zip loader in InvenioRDM core modules to load zipped files, so we can't distribute it zipped right now. We could save a lot of space by doing so if it was supported.

**Pull request merged**